kubernetes_asyncio/watch/watch.py

```diff
@@ -169,12 +177,22 @@ def stream(self, func, *args, **kwargs):
                 if should_stop:
                     watch.stop()
         """
+        if self.resp is not None:
```

Comment:
Can we pack this 3 lines inside a new method close()? We will be able to use it internally and also by end-users in their apps.